source/compression/OP_COMPRESSED.md

```diff
@@ -0,0 +1,432 @@
+
+  mongod should have logged the following:
+
+```
+      {"t":{"$date":"2021-04-08T13:28:38.898-06:00"},"s":"D3", "c":"NETWORK",  "id":22927,   "ctx":"conn2","msg":"Decompressing message","attr":{"compressor":"snappy"}}
+```
+
+The results of the program should have been:
+
+```
+      { "ok" : 1 }
+```
+
+- <mongodb://localhost:27017/?compressors=zlib,snappy>
+
+  mongod should have logged the following:
+
+```
+      {"t":{"$date":"2021-04-08T13:28:38.898-06:00"},"s":"D3", "c":"NETWORK",  "id":22927,   "ctx":"conn2","msg":"Decompressing message","attr":{"compressor":"zlib"}}
+```
+
+The results of the program should have been:
+
+```
+      { "ok" : 1 }
+```
+
+- Create example program that authenticates to the server using SCRAM-SHA-1, then creates another user (MONGODB-CR),
+  then runs hello followed with serverStatus.
+
+- Reconnect to the same server using the created MONGODB-CR credentials. Observe that the only command that was
+  decompressed on the server was `serverStatus`, while the server replied with OP_COMPRESSED for at least the
+  serverStatus command.
+
+## Motivation For Change
+
+Drivers provide the canonical interface to MongoDB. Most tools for MongoDB are written with the aid of MongoDB drivers.
+There exist a lot of tools for MongoDB that import massive datasets which could stand to gain a lot from compression.
+Even day-to-day applications stand to gain from reduced bandwidth utilization at low cpu costs, especially when doing
+large reads off the network.
+
+Not all use cases fit compression, but we will allow users to decide if wire compression is right for them.
+
+## Design rationale
+
+Snappy has minimal cost and provides a reasonable compression ratio, but it is not expected to be available for all
+languages MongoDB Drivers support. Supporting zlib is therefore important to the ecosystem, but for languages that do
+support snappy we expected it to be the default choice. While snappy has no knobs to tune, zlib does have support for
+specifying the compression level (tuned from speed to compression). As we don’t anticipate adding support for
+compression libraries with complex knobs to tune this specification has opted not to define a complex configuration
+structure and only define the currently relevant `zlibCompressionLevel`. When other compression libraries are supported,
+adding support for configuring that library (if any is needed) should be handled on a case by case basis.
+
+More recently, the MongoDB server added Zstandard (zstd) support for another modern alternative to zlib.
+
+## Backwards Compatibility
+
+The new argument provided to the MongoDB Handshake has no backwards compatible implications as servers that do not
+expect it will simply ignore it. This means a server will therefore never reply with a list of acceptable compressors
+which in turns means a client CANNOT use the OP_COMPRESSED opcode.
+
+## Reference Implementation
+
+- [mongoc](https://jira.mongodb.org/browse/CDRIVER-2116)
```

Comment:
I did find it a bit unexpected the link to the reference implementation was to a JIRA ticket with no single linked PR but I guess that's another topic since it linked to that same ticket before.